.github/workflows/translation.yml

```diff
@@ -0,0 +1,275 @@
+##     PAT with repo scope to allow the  ##
+##     action to issue PRs. more info:   ##
+##     https://github.com/peter-evans/create-pull-request#action-inputs
+##   STORAGE_ACCOUNT_NAME: Azure Blob    ##
+##     storage account name              ##
+##   TRANSLATOR_API_KEY: Key for Azure   ##
+##     Cognitive Service Translate API   ##
+##   TRANSLATOR_API_URI: URI of Azure    ##
+##     Document Translator API           ##
+## Expects the folowing variables:       ## 
+##   DOCUMENT_PATH: Path to source       ##
+##     document for translation          ##
+###########################################
+###########################################
+name: Translate Document
+
+on:
+  workflow_dispatch:
+  
+permissions:
+  id-token: write
+  contents: read
+
+env:
+  INPUT_CONTAINER_NAME: translatesource
+  GLOSSARY_CONTAINER_NAME: translateglossaries
+
+jobs:
+  ###################################
+  # Upload source document to Azure #
+  ###################################
+  upload-source:
+    name: Upload source document to Azure
+    runs-on: ubuntu-latest
+    steps:
+      ##########################
+      # Checkout the code base #
+      ##########################
+      - name: Checkout Code
+        uses: actions/checkout@v3
+      ##########################
+      # Login to Azure         #
+      ##########################
+      - name: "Az CLI login"
+        uses: azure/login@v1
+        with:
+          client-id: ${{ secrets.AZURE_CLIENT_ID }}
+          tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+          subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+      ##############################
+      # Create the input container #
+      ##############################
+      - name: "Create input container with Azure CLI"
+        run: |
+          set -eux
+          az storage container create \
+            --name $INPUT_CONTAINER_NAME \
+            --account-name ${{ secrets.STORAGE_ACCOUNT_NAME }} \
+            --auth-mode login
+
+          az storage container create \
+            --name $GLOSSARY_CONTAINER_NAME \
+            --account-name ${{ secrets.STORAGE_ACCOUNT_NAME }} \
+            --auth-mode login
+      ##############################
+      # Upload the source document #
+      ##############################
+      - name: "Upload source document with Azure CLI"
+        run: |
+          set -eux
+          end=`date -u -d "30 minutes" '+%Y-%m-%dT%H:%MZ'`
+          sas=`az storage container generate-sas -n $INPUT_CONTAINER_NAME --auth-mode login --as-user --account-name ${{ secrets.STORAGE_ACCOUNT_NAME }} --https-only --permissions dlrw --expiry $end -o tsv`
+          az storage blob upload \
+            --overwrite true \
+            --account-name ${{ secrets.STORAGE_ACCOUNT_NAME }} \
+            --container-name $INPUT_CONTAINER_NAME \
+            --name ${{ vars.DOCUMENT_PATH }} \
+            --file ${{ vars.DOCUMENT_PATH }} \
+            --sas-token $sas
```

Comment:
Can be a reusable workflow for upload x to y

Reply:
fixed so there is no need to